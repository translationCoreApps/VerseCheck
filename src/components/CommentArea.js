import React from 'react';
import {FormGroup, FormControl, Glyphicon} from 'react-bootstrap';
import style from '../css/Style';

let CommentArea = (props) => {
  return (
    <div style={style.commentArea}>
      <div style={{fontWeight: 'bold'}}>
        <Glyphicon glyph='comment' style={{marginRight: '5px'}} />
        Comment
      </div>
      <FormGroup style={{flex: 'auto', display: 'flex' }} controlId="formControlsTextarea">
        <FormControl autoFocus
          componentClass='textarea'
          type='text'
          defaultValue={props.comment}
          style={{flex: 'auto'}}
          onBlur={props.actions.handleComment.bind(this)}
          onInput={props.actions.checkComment.bind(this)}
        />
      </FormGroup>
    </div>
  );
};

module.exports = CommentArea;
