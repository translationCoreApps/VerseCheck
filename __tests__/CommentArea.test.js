/* eslint-env jest */
import React from 'react';
import CommentArea from '../src/components/CommentArea';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CommentArea component Tests', () => {
  let props;

  beforeEach(()=> {
    props = {
      actions: {
        handleComment: jest.fn(),
        checkComment: jest.fn(),
      },
      comment: "my comment"
    };
  });

  test('Check CommentArea component', () => {
    const wrapper = shallow(<CommentArea {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.text()).toEqual('<Glyphicon />Comment<FormGroup />');
  });
});
