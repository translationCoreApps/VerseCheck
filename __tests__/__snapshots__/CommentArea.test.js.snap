// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentArea component Tests Check CommentArea component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "auto",
      "flexDirection": "column",
      "height": "100%",
      "padding": "5px 15px 0 15px",
    }
  }
>
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="comment"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    />
    Comment
  </div>
  <FormGroup
    bsClass="form-group"
    controlId="formControlsTextarea"
    style={
      Object {
        "display": "flex",
        "flex": "auto",
      }
    }
  >
    <FormControl
      autoFocus={true}
      bsClass="form-control"
      componentClass="textarea"
      defaultValue="my comment"
      onBlur={[Function]}
      onInput={[Function]}
      style={
        Object {
          "flex": "auto",
        }
      }
      type="text"
    />
  </FormGroup>
</div>
`;
